hudi-client/src/main/java/org/apache/hudi/io/HoodieCommitArchiveLog.java

```diff
@@ -249,6 +249,13 @@ public void archive(List<HoodieInstant> instants) throws HoodieCommitException {
       LOG.info("Wrapper schema " + wrapperSchema.toString());
       List<IndexedRecord> records = new ArrayList<>();
       for (HoodieInstant hoodieInstant : instants) {
+
+        // filter empty instant, like *.commit.requested
+        byte[] instantDetails = commitTimeline.getInstantDetails(hoodieInstant).get();
+        if (instantDetails == null || instantDetails.length == 0) {
+          continue;
+        }
+
```

Comment:
@lamber-ken : We did not catch this in our unit-tests because I had inadvertently created requested files with proper commit metadata for test data generation. Please look at HoodieTestDataGenerator.createCommitFile(). Can you also fix this as part of this PR so that only empty requested commit files are present ?

Reply:
Yeah, will rich unit-tests.

Reply:
@lamber-ken : This is not directly related but I created https://jira.apache.org/jira/browse/HUDI-483 for tracking this unit test issue. Please assign it to yourself if you are planning to work on this

Reply:
@bvaradar Sorry, I probably missed this notification. IMO, it is not logic error, unit tests should not cover it, because we don't create an empty file manually. WDYT?